### ISSUE TITLE
Shibboleth: Cannot convert a builtin account to shib, does not accept password.

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#3119]
Logging in via Shib using either Harvard or testshib on internal I can log in if there is no preexisting local account with the same email address. If there is a preexisting email, I need to convert the account on log in and that is currently failing with a bad password message, though the password is correct.

@pdurbin
This functionality was working when tested on internal during the QA phase. However, some minor tweaks were made during UX phase and there were some resolved merge conflicts.

**Comment by pdurbin** [comment]
@kcondon I just tried this on https://shibtest.dataverse.org running v. 4.3.1 build develop-7fe7040 (the latest code in "develop" as of this writing) and converting a builtin account worked fine. Here are some screenshots:

![shib_-_2016-05-13_16 32 11](https://cloud.githubusercontent.com/assets/21006/15261093/7bca872c-1928-11e6-991a-44131a7eeb5a.png)

![account_-_root_dataverse_-_2016-05-13_16 32 23](https://cloud.githubusercontent.com/assets/21006/15261104/8bcb2a6e-1928-11e6-9a0e-2a46923a212e.png)

Here's what it showed in server.log (I must have it cranked up to fine):

```
[2016-05-13T16:32:15.365-0400] [glassfish 4.1] [FINE] [] [edu.harvard.iq.dataverse.Shib] [tid: _ThreadID=51 _ThreadName=jk-connector(2)] [timeMillis: 1463171535365] [levelValue: 500] [CLASSNAME: edu.harvard.iq.dataverse.Shib] [METHODNAME: confirmAndConvertAccount] [[
  builtin username: philip_durbin]]

[2016-05-13T16:32:15.475-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean] [tid: _ThreadID=51 _ThreadName=jk-connector(2)] [timeMillis: 1463171535475] [levelValue: 800] [[
  converting user 8 from builtin to shib]]

[2016-05-13T16:32:15.476-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean] [tid: _ThreadID=51 _ThreadName=jk-connector(2)] [timeMillis: 1463171535476] [levelValue: 800] [[
  builtin user identifier: @philip_durbin]]

[2016-05-13T16:32:15.478-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean] [tid: _ThreadID=51 _ThreadName=jk-connector(2)] [timeMillis: 1463171535478] [levelValue: 800] [[
  we expect this to be 'builtin': builtin]]

[2016-05-13T16:32:15.478-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean] [tid: _ThreadID=51 _ThreadName=jk-connector(2)] [timeMillis: 1463171535478] [levelValue: 800] [[
  this should be 'pete' or whatever the old builtin username was: philip_durbin]]

[2016-05-13T16:32:15.487-0400] [glassfish 4.1] [FINE] [] [edu.harvard.iq.dataverse.Shib] [tid: _ThreadID=51 _ThreadName=jk-connector(2)] [timeMillis: 1463171535487] [levelValue: 500] [CLASSNAME: edu.harvard.iq.dataverse.Shib] [METHODNAME: logInUserAndSetShibAttributes] [[
  Groups for user 8 (@philip_durbin): [All Users (:AllUsers), Authenticated Users (:authenticated-users)]]]
```

I'm heading out for the day, but I'm passing it back to you for consideration, @kcondon

**Comment by kcondon** [comment]
OK, thanks. We need to get this working on the test box before we can mark this as resolved.

**Comment by pdurbin** [comment]
On dataverse-internal running v. 4.3.1 build 95-68c7239 (latest from 2469-widgets branch), I'm able to replicate the problem of not being prompted to convert my account from local to shib. My first guess is that the leading spaces persisted in the database for my email address (see #2945 and #3044) are the problem:

```
[root@dvn-vm4 ~]# curl -s -H "X-Dataverse-key: $API_TOKEN" http://localhost:8080/api/admin/authenticatedUsers | jq . | grep durbin@
      "email": "  philip_durbin@harvard.edu",
```

**Comment by pdurbin** [comment]
Sure enough, after I eliminated the leading whitespace from my `"  philip_durbin@harvard.edu"` (which I was able to do as my own user through the GUI on the account page), I was able to successfully convert my account on dataverse-internal running v. 4.3.1 build 95-68c7239:

![shib_-_2016-05-16_12 36 42](https://cloud.githubusercontent.com/assets/21006/15296385/3457b1de-1b63-11e6-989b-1b7fdad4bdbc.png)

![account_-_root_dataverse_-_2016-05-16_12 37 04](https://cloud.githubusercontent.com/assets/21006/15296384/3456875a-1b63-11e6-8a0d-da7f1e020332.png)

@kcondon I'm going to pass this back to you for consideration. #2945 was already closed as a duplicate of #3044. Perhaps #3044 should be worked on before we release 4.4.

**Comment by kcondon** [comment]
Was no longer able to reproduce, closing.
